File: backend/src/main/java/com/example/aichat/service/ChatService.java
package com.example.aichat.service;

import com.example.aichat.client.AIClient;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;

@Service
public class ChatService {

    private final AIClient aiClient;

    public ChatService(AIClient aiClient) {
        this.aiClient = aiClient;
    }

    public Flux<String> streamChat(String message, String agentId,
                                    String conversationId, String messageId) {
        return aiClient.streamGenerate(message, agentId, conversationId, messageId);
    }

    public reactor.core.publisher.Mono<String> getChatHistory(String conversationId) {
        return aiClient.getChatHistory(conversationId);
    }
}

File: backend/src/main/kotlin/com/example/aichat/service/ChatService.kt
package com.example.aichat.service

import com.example.aichat.client.AIClient
import org.springframework.stereotype.Service
import reactor.core.publisher.Flux

@Service
class ChatService(
    private val aiClient: AIClient
) {

    fun streamChat(message: String, conversationId: String?): Flux<String> {
        return aiClient.streamGenerate(message, conversationId)
            .map { response -> response.token }
    }
}
